### ISSUE TITLE
Colormaps for wave parameters

### ISSUE THREAD

**Issue by caiostringari** [matplotlib/cmocean#22]
Great work on the colormaps !

Are you planning to add colormaps for wave parameters such as wave height, period and direction ?

Thanks

**Comment by kthyng** [comment]
I'd be willing to work on some colormaps on that. Do you have anything in
mind? I haven't worked much with wave information. I would suggest a
circular colormap for wave direction so there wouldn't be a jump between 0
and 360 degrees.

On Sat, Sep 12, 2015 at 3:24 PM, Caio Eadi Stringari <
notifications@github.com> wrote:

> Great work on the colormaps !
>
> Are you planning to add colormaps for wave parameters such as wave height,
> period and direction ?
>
> Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/cmocean/issues/22>.
>



-- 
Kristen M. Thyng
Assistant Research Scientist
Department of Oceanography
Texas A&M University
Eller O&M 608
http://kristenthyng.com

**Comment by caiostringari** [comment]
I`ve been using RdYlBu_r for wave height and RdYlGn_r for wave period, but because they are diverging colormaps they get a bit to "whitish" in the center of the scale. For wave direction, the only circular scheme I can think about is (the not so great) HSV. One good thing is that a colormap for wave directions will work for tidal phase as well... 

The following plots show how the RdYlBu_r and RdYlGn_r colormaps look when plotting heights and periods.

![ww hs 2015050800z](https://cloud.githubusercontent.com/assets/8363752/9839907/702866bc-5a5d-11e5-8985-03238098a1a2.png)
![ww tp 2015050800z](https://cloud.githubusercontent.com/assets/8363752/9839906/7027ae8e-5a5d-11e5-9be2-d9c82ca5aedf.png)

**Comment by kthyng** [comment]
Yeah I think you're right that sequential colormaps would be best for
significant wave height and wave peak period. What do you think about using
the positive half of the Freesurface colormap for significant wave height,
in the same manner that speed is the positive half of the velocity
colormap? Or are there times when you or others would want these to be
distinct?

With an analogous "similar units" argument, we could maybe use half of the
vorticity colormap for wave peak period (1/s vs. s) though this seems like
a bigger stretch to me, and moreover I don't think the reds from the
positive half of the free surface would look particularly good alongside
either half of the vorticity colormap (maybe the turquoise side would be
ok). Since these two are plotted together a lot, I think they should be
complementary.

I haven't really used the uniform lightness, 0 to 360 hue change circular
colormaps but I'm interested in checking a few out. I'll search around and
let you know what I find.

On Sun, Sep 13, 2015 at 7:36 PM, Caio Eadi Stringari <
notifications@github.com> wrote:

> I`ve been using RdYlBu_r for wave height and RdYlGn_r for wave period, but
> because they are diverging colormaps they get a bit to "whitish" in the
> center of the scale. For wave direction, the only circular scheme I can
> think about is (the not so great) HSV. One good thing is that a colormap
> for wave directions will work for tidal phase as well...
>
> The following plots show how the RdYlBu_r and RdYlGn_r colormaps look when
> plotting heights and periods.
>
> [image: ww hs 2015050800z]
> <https://cloud.githubusercontent.com/assets/8363752/9839907/702866bc-5a5d-11e5-8985-03238098a1a2.png>
> [image: ww tp 2015050800z]
> <https://cloud.githubusercontent.com/assets/8363752/9839906/7027ae8e-5a5d-11e5-9be2-d9c82ca5aedf.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/cmocean/issues/22#issuecomment-139931119>.
>



-- 
Kristen M. Thyng
Assistant Research Scientist
Department of Oceanography
Texas A&M University
Eller O&M 608
http://kristenthyng.com

**Comment by kthyng** [comment]
I've been playing around with viscm to make a circular colormap. I've
committed what I came up with in my fork of cmocean (
https://github.com/kthyng/cmocean) — want to check it out and see what you
think? It is called "Phase." I tried avoiding using both reds and greens to
avoid color blindness issues, though I found that the colorblind versions
of these circular colormaps looked pretty similar whether I avoided those
areas or not, so I ended up using all the colors.

On Fri, Sep 25, 2015 at 2:23 PM, Kristen Thyng <kthyng@gmail.com> wrote:

> Yeah I think you're right that sequential colormaps would be best for
> significant wave height and wave peak period. What do you think about using
> the positive half of the Freesurface colormap for significant wave height,
> in the same manner that speed is the positive half of the velocity
> colormap? Or are there times when you or others would want these to be
> distinct?
>
> With an analogous "similar units" argument, we could maybe use half of the
> vorticity colormap for wave peak period (1/s vs. s) though this seems like
> a bigger stretch to me, and moreover I don't think the reds from the
> positive half of the free surface would look particularly good alongside
> either half of the vorticity colormap (maybe the turquoise side would be
> ok). Since these two are plotted together a lot, I think they should be
> complementary.
>
> I haven't really used the uniform lightness, 0 to 360 hue change circular
> colormaps but I'm interested in checking a few out. I'll search around and
> let you know what I find.
>
> On Sun, Sep 13, 2015 at 7:36 PM, Caio Eadi Stringari <
> notifications@github.com> wrote:
>
>> I`ve been using RdYlBu_r for wave height and RdYlGn_r for wave period,
>> but because they are diverging colormaps they get a bit to "whitish" in the
>> center of the scale. For wave direction, the only circular scheme I can
>> think about is (the not so great) HSV. One good thing is that a colormap
>> for wave directions will work for tidal phase as well...
>>
>> The following plots show how the RdYlBu_r and RdYlGn_r colormaps look
>> when plotting heights and periods.
>>
>> [image: ww hs 2015050800z]
>> <https://cloud.githubusercontent.com/assets/8363752/9839907/702866bc-5a5d-11e5-8985-03238098a1a2.png>
>> [image: ww tp 2015050800z]
>> <https://cloud.githubusercontent.com/assets/8363752/9839906/7027ae8e-5a5d-11e5-9be2-d9c82ca5aedf.png>
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/matplotlib/cmocean/issues/22#issuecomment-139931119>.
>>
>
>
>
> --
> Kristen M. Thyng
> Assistant Research Scientist
> Department of Oceanography
> Texas A&M University
> Eller O&M 608
> http://kristenthyng.com
>



-- 
Kristen M. Thyng
Assistant Research Scientist
Department of Oceanography
Texas A&M University
Eller O&M 608
http://kristenthyng.com

**Comment by caiostringari** [comment]
I liked your suggestions for the wave heights, I've used a similar colors for this variable with good results before. For the wave period, I guess either the first or the second half of the vorticity colormap would work fine. I used modified version of the Blues colormap for wave periods before and people seemed to have liked it.

Regarding the circular colormap, I would just make the blue part a little darker.

Sometimes, when forecasting especially, a distinct colormarp for the wave heights makes the life of forecasters a lot easier because them can quickly associate a color with a specific sea state. But I really don't know how this fits here...
